src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/CSharpIsEventBinder.cs

```diff
@@ -22,7 +22,17 @@ public Expr DispatchPayload(RuntimeBinder runtimeBinder, ArgumentObject[] argume
 
         [RequiresUnreferencedCode(Binder.TrimmerWarning)]
         public void PopulateSymbolTableWithName(Type callingType, ArgumentObject[] arguments)
-            => SymbolTable.PopulateSymbolTableWithName(Name, null, arguments[0].Info.IsStaticType ? arguments[0].Value as Type : arguments[0].Type);
+        {
+            var value = arguments[0].Value as Type;
+            if (arguments[0].Info.IsStaticType)
+            {
+                if (value == null)
+                    throw new Exception("invalid cast arguments[0].Value to Type");
+                SymbolTable.PopulateSymbolTableWithName(Name, null, value);
+            }
+            else
+                SymbolTable.PopulateSymbolTableWithName(Name, null, arguments[0].Type);
```

Comment:
Can you demonstrate in which condition the invalid cast branch will be hit?
If an invalid cast exception should be thrown, changing `as` to `(Type)arguments[0].Value` should be better.

Reply:
We can't analyze the entire chain of calls, but this code refers to a variable resulting from a conversion without first checking whether the conversion was successful

Reply:
There can be conventions for states of `ArgumentObject`. The better approach should be expressing the convention more in code.

Reply:
In addition, such check can be covered by nullable reference type analysis, since cast result of `as` is nullable.

Currently NRT warnings are disabled for `Microsoft.CSharp`. /cc @stephentoub Should we take effort to enable it?